### PR TITLE
Add user signup and login, complete with password reset

### PR DESCRIPTION
**Pull request by hancush**
This PR reverts the revert of @evz's login changes. I've removed some obsolete/redundant code on top of his changes, and it's ready now for password reset implementation.

### CODE REVIEW

**Comment by fgregg** [comment]
- This needs a password reset option. It's enough to add it to the modal.  
- We also need to be cognizant of any interactions with caching.
- I have mandrill already set up to send email from bettergov.org. We will use the email address no-reply@bettergov.org

@evz

**Comment by evz** [comment]
@hancush @fgregg The basic scaffolding for this is together. The main thing that needs to be edited is the language and layout for the templates that are used for the various parts of the reset loop. I'd love to sit with someone and hammer that out.

**Comment by evz** [comment]
@fgregg @hancush This is ready for review. I think the next step is to get it deployed so we can get some feedback on the way it works.

File: payroll/forms.py [hunk 1]
```diff
@@ -0,0 +1,123 @@
...
+
+        user.userzipcode = userzipcode
+
+        user.save()
+
+        self.add_to_salsa(user)
+
+        return user
+
+    def add_to_salsa(self, user):
```
[the remark this answers is not in the document]
Comment (evz) [reply payroll/forms.py]: Hmm, now that I think about it, this is an open source project now, right? Might kinda suck to have their mailing list get bombed with a bunch of junk subscribers. I'll add this dict to local settings.

[the remark this answers is not in the document]
Comment (hancush) [reply payroll/forms.py]: great idea!

File: payroll/views.py [hunk 2]
```diff
@@ -541,15 +548,31 @@ class SearchView(ListView, PayrollSearchMixin, FacetingMixin):
...
 
-        except DisallowedSearchException:
+            except DisallowedSearchException:
+                self.allowed = False
+                results = []
+        else:
             self.allowed = False
-            return []
+            results = []
+            self.facets = {}
```
Comment (hancush) [inline payroll/views.py]: this line is redundant now.

**Pull request merged**